zerver/lib/actions.py

```diff
@@ -2379,11 +2379,12 @@ def check_message(sender: UserProfile, client: Client, addressee: Addressee,
         recipient = stream.recipient
 
         # This will raise JsonableError if there are problems.
-        validate_sender_can_write_to_stream(
-            sender=sender,
-            stream=stream,
-            forwarder_user_profile=forwarder_user_profile
-        )
+        if sender.bot_type != sender.OUTGOING_WEBHOOK_BOT:
```

Comment:
I'm afraid that just checking the bot type is not sufficient here.  We can only allow outgoing webhooks to reply to messages in the very specific case that we are dealing with a message reply from an outgoing webhook bot user who's running a server that we contacted with an HTTP request via the outgoing-webhook codepath.  So the proper fix here is to pass in a flag from the appropriate place in `zerver/lib/outgoing_webhook.py`.

Reply:
Okay!